pkg/scheduler/preemption/preemption.go

```diff
@@ -438,6 +438,10 @@ func (p *Preemptor) findCandidates(wl *kueue.Workload, cq *cache.ClusterQueueSna
 				continue
 			}
 
+			if meta.IsStatusConditionTrue(candidateWl.Obj.Status.Conditions, kueue.WorkloadEvicted) {
```

Comment:
I don’t claim :100: expertise in preemption, but my understanding is that preemption candidates are selected from *currently admitted* workloads. If that’s correct, then this condition should never be met.

Also, since preemption nomination and actual preemption happen in two distinct phases, there shouldn’t be any collision between the “nomination” and “scheduling” phases. Given that, it’s not clear to me how this check helps or what specific scenario it is intended to guard against.

Could you please clarify?

Reply:
> I don’t claim 💯 expertise in preemption, but my understanding is that preemption candidates are selected from currently admitted workloads. If that’s correct, then this condition should never be met.

I believe in this case, the workload is evicted but still considered admitted until eviction is done. So this captures workloads that are basically in the process of being evicted